### PR TITLE
Fix indexing on axis keys using generic map (related to spatial averaging)

### LINKED ISSUE #163: spatial_average not interpreting latitude
### What versions of software are you using?

-   Package Version: Main branch / Release 0.1.0

### What are the steps to reproduce this issue?
```
p = '/p/css03/esgf_publish/CMIP6/CMIP/UA/MCM-UA-1-0/historical/r1i1p1f1/Amon/tas/gn/v20190731/*nc' 
ds = xcdat.open_mfdataset(p, data_var='tas')
ds.xcdat.spatial_avg()
```

> KeyError: "The data variable 'tas' is missing a 'lat' dimension, which is required for spatial averaging."

### What were you expecting to happen?
```
print(ds)
```

> Dimensions:    (time: 1980, latitude: 80, longitude: 96, bnds: 2)
> Coordinates:
>   * bnds       (bnds) float64 1.0 2.0
>   * longitude  (longitude) float64 0.0 3.75 7.5 11.25 ... 348.8 352.5 356.2
>   * latitude   (latitude) float64 -88.29 -86.07 -83.84 ... 83.84 86.07 88.29
>   * time       (time) object 1850-01-17 00:00:00 ... 2014-12-17 00:00:00

The file includes a `latitude` coordinate instead of longitude, but I would have expected it to interpret `lat`.

### PR DESCRIPTION
**Pull request by tomvothecoder**
## Description

- Closes #163 

Summary of Changes
- Add `GENERIC_AXES_MAP` for mapping CF compliant keys to their generic representations
- Add `_get_generic_axis_keys()` for proper indexing in spatial averaging methods
- Update bounds methods to use `GENERIC_AXES_MAP`
- Update incorrect axis refs to axis

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

If applicable:

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass with my changes (locally and CI/CD build)
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have noted that this is a breaking change for a major release (fix or feature that would cause existing functionality to not work as expected) -- will include notes in next minor release instead

### CODE REVIEW

**Comment by tomvothecoder** [comment]
Hi @pochedls, this PR is ready for review. 

It fixes #163, `KeyError: "The data variable 'tas' is missing a 'lat' dimension, which is required for spatial averaging."`. Basically, your example datasets use the CF long key `"latitude"` instead of the short key `"lat"` without a proper mapping of `"lat"` to `"latitude"`. CF-xarray can't interpret `"lat"` as being `"latitude"` when retrieving the coordinates and bounds, so it fails.

My solution is to use a map of CF long and short keys to their generic versions for a more robust retrieval of coordinates. For example, "lat" and "latitude" maps to "Y", which CF-xarray can interpret agnostically if the CF attribute `Axis: "Y"` is set.

File: xcdat/bounds.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ class BoundsAccessor:
     >>> lon_bounds = ds.bounds.get_bounds("lon") # or pass "longitude"
     >>> time_bounds = ds.bounds.get_bounds("time")
 
-    Add coordinate bounds for a specific axis if they don't exist:
+    Add coordinate bounds for a specific axes if they don't exist:
```
Comment (pochedls) [inline xcdat/bounds.py:43]: Should this be "axis" (singular)?

**Review by pochedls — APPROVED** [review]
Hi Tom - this works and looks good to me. I didn't understand why you swapped axis / axes - it seemed like it may have made more sense before (looping over each axis (singular) in axes (plural)).

**Comment by tomvothecoder** [comment]
Thanks Steve, I think I got carried away with my control and replace and forgot the definition of axis/axes again for the 10th time.

**Pull request merged**